osu.Game/Database/RealmContextFactory.cs

```diff
@@ -106,6 +107,8 @@ public RealmContextFactory(Storage storage, string filename, IDatabaseContextFac
 
         private void cleanupPendingDeletions()
         {
+            new RealmFileStore(this, storage).Cleanup();
```

Comment:
Isn't the purpose of the block below to remove backlinks to the files? At a glance it seems like this should be the very last thing run.

Reply:
Could be argued so, yeah. I intentionally ordered this way to only clean up files on the next startup (in case the user does something weird and tries to recover by reloading the files back, which would avoid a copy) but maybe it makes more sense to match the existing behaviour?